File: packages/gbnf-experiment/data/20260909T190508Z_7c1544c0/ported_implementation/gbnf/utils/errors/get_input_as_string.py
from typing import List, Union

ValidInput = Union[str, int, List[int]]


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src
    code_points = src if isinstance(src, list) else [src]
    return ''.join(chr(cp) for cp in code_points)

File: proposed-projects/round-trip-experiment/reverse/20260909T040730Z_0947f229/20260910T065554Z_5d586b52/ported_implementation/src/utils/errors/get-input-as-string.ts
/** Port of `gbnf/utils/errors/get_input_as_string.py`. */

import type { ValidInput } from '../../grammar-graph/types.js';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  const codePoints = Array.isArray(src) ? src : [src];
  return codePoints.map((cp) => String.fromCodePoint(cp)).join('');
};
